docs/core/diagnostics/metrics-collection.md

```diff
@@ -380,40 +256,37 @@ static void OnMeasurementRecorded<T>(Instrument instrument, T measurement, ReadO
-of measurement will be received by the callback. For example, a `Counter<int>` generates `int` measurements whereas a
-`Counter<double>` generates `double` measurements. Instruments can be created with `byte`, `short`, `int`, `long`,
-`float`, `double`, and `decimal` types. We recommend registering a callback for every data type unless you have scenario-specific knowledge that not all data types will be needed, such as in this example. Making repeated calls to
-SetMeasurementEventCallback() with different generic arguments may appear a little unusual. The API is designed this way
-to allow MeterListeners to receive measurements with extremely low performance overhead, typically just a few nanoseconds.
-
-When MeterListener.EnableMeasurementEvents() was called initially, there was an opportunity to provide a `state` object as
-one of the parameters. That object can be anything you want. If you provide a state object in that call, then it will be
-stored with that instrument and returned to you as the `state` parameter in the callback. This is intended both as a
-convenience and as a performance optimization. Often listeners need to create an object for each instrument that will
-store measurements in memory and have code to do calculations on those measurements. Although you could create a Dictionary
-that maps from the instrument to the storage object and look it up on every measurement, that would be much slower than
-accessing it from `state`.
+:::code language="csharp" source="snippets/Metrics/Program.cs" id="snippet_sme" highlight="1,15-99":::
+
+The generic parameter controls which data type of measurement is received by the callback. For example, a `Counter<int>` generates `int` measurements, `Counter<double>` generates `double` measurements. Instruments can be created with `byte`, `short`, `int`, `long`, `float`, `double`, and `decimal` types. We recommend registering a callback for every data type unless you have scenario-specific knowledge that not all data types are needed. Making repeated calls to `SetMeasurementEventCallback` with different generic arguments may appear a little unusual. The API was designed this way to allow `MeterListeners` to receive measurements with  low performance overhead, typically just a few nanoseconds.
+
+When `MeterListener.EnableMeasurementEvents` is called, a `state` object can be provided as
+one of the parameters. The `state` object is arbitrary. If you provide a state object in that call, then it is stored with that instrument and returned to as the `state` parameter in the callback. This is intended both as a convenience and as a performance optimization. Often listeners need to:
+
+- Create an object for each instrument that is storing measurements in memory.
+- Have code to do calculations on those measurements.
+
+<!--Why are we explaining low perf options? Metrics should be ultra low cost. Can we remove this? Folks that need this approach are smart enough to figure it out. -->
```

Comment:
In the original text it was present to contrast better and worse performing options. Its np if you delete it to increase brevity.